wp-cache-export.php

```diff
@@ -0,0 +1,472 @@
+    if ( $this->backupFileExists() )
+      $renamed = @rename( self::$cache_config_file_backup, self::$cache_config_file );
+    if ( ! $renamed ) {
+      wp_safe_redirect( add_query_arg( 'message', 4, $location ) );
+      exit;
+    }
+    foreach ( get_option( '_wp_super_cache_backup_options', array() ) as $key => $value ) {
+        update_option( $key, $value );
+    }
+    delete_option( '_wp_super_cache_backup_options' );
+    include self::$cache_config_file;
+    wp_safe_redirect( add_query_arg( 'message', 5, $location ) );
+    exit;
+  }
+
+  /**
+   * Removes the backup file and the database backup option.
+   *
+   * If the import is successful and the user does not want to have their back ups any longer then this function will
+   * remove the wp-cache-config-backup.php file and the _wp_super_cache_backup_options database option.
+   * The backs ups are no longer retrievable after this.
+   *
+   * @since  1.4.4
+   *
+   * @uses  wp_safe_redirect
+   * @uses  delete_option
+   *
+   * @return  nothing The user is redirected to the import/export form after the removal of the backup file and database option.
+   */
+
+  public function remove() {
+    if ( ! $this->can_remove() ) {
+      return;
+    }
+    $location = add_query_arg( 'tab', 'export', admin_url( 'options-general.php?page=wpsupercache' ) );
+    if ( $this->backupFileExists() )
+      $file = @unlink( self::$cache_config_file_backup );
+    if ( ! $file ) {
+      wp_safe_redirect( add_query_arg( 'message', 4, $location ) );
+      exit;
+    }
+    delete_option( '_wp_super_cache_backup_options' );
+    wp_safe_redirect( add_query_arg( 'message', 6, $location ) );
+    exit;
+  }
+
+  /**
+   * This is a private function that gathers all the database options WP Super Cache stores outside of the wp-cache-config.php file.
+   * The array of option names is stored as a class constant.
+   *
+   * @since  1.4.4
+   *
+   * @uses  get_option Gather the array of option key/value pairs.
+   *
+   * @return array An array of the key/value pairs for the WP Super Cache options stored in the database
+   */
+  private function gather_plugin_options() {
+    $options = array();
+    foreach ( self::OPTIONS  as $value ) {
+      $options[ $value ] = get_option( $value );
+    }
+    return array( '_wp_super_cache_options' => $options );
+  }
+
+  /**
+   * This is a private function dedicated to sanitizing the JSON file input.
+   * Given that a malicious JSON file could be uploaded and converted into the plugin settings it is best practice to sanitize
+   * the JSON values before converting them.
+   *
+   * Since only numeric values and strings are accepted inputs we cast integers on numeric values and strip tags/escape
+   * the html of string values.
+   *
+   * @since  1.4.4
+   *
+   * @uses esc_html Escape the string values
+   *
+   * @return number|string A sanitized version of the input value.
+   */
+
+  private function sanitize_value( $value, $setting ) {
```

Comment:
This was a legacy function from before the WP_Super_Cache_Sanitizer was coded. Good catch b842dbf90.